### PR TITLE
Enable http redirection on deck

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**
Change the ingress class to nginx, enable ssl redirection. 

**Related issue(s)**
Implements #85

### CODE REVIEW

**Comment by Demonsthere** [comment]
Keep in mind we have to update the manifest on the cluster after merging this

**Comment by Demonsthere** [comment]
I have created an extra ingress object on our cluster, and set it to use nginx + force https. 
You can see the behaviour here: 35.242.246.53

**Comment by Demonsthere** [comment]
There seems to be a bug in 0.20 which causes frequent pod restarts, so I bumped it to 0.21

**Comment by michal-hudy** [comment]
@Demonsthere @suleymanakbas91 Guys, please don't change production cluster configuration before PR is merged...

**Comment by Demonsthere** [comment]
@michal-hudy I didn't change the production ingress configuration, just added a new one, using the nginx ingress

File: development/install-prow.sh [hunk 1]
```diff
@@ -73,6 +73,32 @@ go run "${CURRENT_DIR}/tools/cmd/secretspopulator/main.go" --project="${PROJECT}
...
 kubectl apply -f "${PROW_CLUSTER_DIR}/starter.yaml"
 
-# Add annotations to Prow Ingress 
-kubectl annotate ingress ing kubernetes.io/ingress.class=nginx nginx.ingress.kubernetes.io/ssl-redirect=false
-kubectl patch ingress ing --type=json -p='[{"op": "replace", "path": "/spec/rules/0/http/paths/0/path", "value":"/"}]'
+# Remove default ingress
+kubectl delete ingress ing
+
+# Recreate ingress with nginx labels
+cat << EOF | kubectl apply -f -
```
Comment (michal-hudy) [inline development/install-prow.sh]: Just update `starter.yaml`

File: prow/cluster/03-tls-ing_ingress.yaml [hunk 2]
```diff
@@ -10,7 +10,8 @@ metadata:
     kubernetes.io/tls-acme: "true"
     certmanager.k8s.io/cluster-issuer: letsencrypt-prod
     certmanager.k8s.io/acme-http01-edit-in-place: "true"
-    kubernetes.io/ingress.class: "gce"
+    kubernetes.io/ingress.class: "nginx"
```
Comment (michal-hudy) [inline prow/cluster/03-tls-ing_ingress.yaml]: `kubernetes.io/ingress.global-static-ip-name` is not supported by `nginx`

**Comment by Demonsthere** [comment]
UPDATE:
I found out that the `deck` element of prow has http->https redirection baked in the very [application](https://github.com/kubernetes/test-infra/blob/f6ca83735e5f785297d092de659e7c1daaecd04f/prow/cmd/deck/main.go#L107) and this can be set in the [deployment](https://github.com/kubernetes/test-infra/blob/f6ca83735e5f785297d092de659e7c1daaecd04f/prow/cluster/deck_deployment.yaml#L47)
With this we can still use the GCP ingress.

File: prow/cluster/README.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ This folder contains files related to the configuration of the Prow production c
 The structure of the folder looks as follows:
 
 ```
+  ├── 00-deck-patch.yaml                # Patch file for the deck deployment used to enable HTTP->HTTPS redirection.
```
Comment (kazydek) [inline prow/cluster/README.md]:
```suggestion
  ├── 00-deck-patch.yaml                # The patch file for the Deck Deployment used to enable HTTP to HTTPS redirection. 	
```

Comment (kazydek) [reply prow/cluster/README.md]: `00`? :)

Comment (Demonsthere) [reply prow/cluster/README.md]: Yes, this scripts should be run first, as it modifies the prow installation deployments, so I could either mark it as `00` or as `01` and bump the rest of scripts by 1 :)

Comment (kazydek) [reply prow/cluster/README.md]: Yup, I was just wondering why you chose the first option ;)

**Pull request merged**